src/toast/todmap/conviqt.py

```diff
@@ -573,31 +566,45 @@ def _get_detectors(self, data):
         all_dets = self._comm.bcast(dets, root=0)
         return all_dets
 
-    def _get_hwpangle(self, tod):
+
+    def _get_hwpangle (self, data   ):
         """
         Return  the spinning HWP angle and the multiplicative
         factor to be applied when added to the  signal polarization angle.
         """
         hwpang = None
-        try:
+        all_hwpang =[]
+        for obs in data.obs:
+            tod= obs["tod"]
+            nsamp = tod.local_samples[1]
             hwpang = tod.local_hwp_angle()
-        except:
-            pass
-        return hwpang
+            if hwpang is not  None :
+                all_hwpang .append(hwpang)
+            else :
+                return hwpang
 
-    def _get_psipol(self, focalplane, det):
+        return  np.hstack (all_hwpang )
+
+
+    def _get_psipol(self, data,  det):
```

Comment:
This implementation will fail if the detector in question is not available in all observations. Please see how `get_pointing` avoids this issue.